### DISCUSSION TITLE
What should I do if I want to recover the parsed data again?

### DISCUSSION THREAD

**Discussion by rjcnd105** [open-circle/valibot#112] [Q&A] [answered]
For example, in @effect/schema, i can decode and encode as follows.

https://github.com/Effect-TS/schema#transformations

I want to use valibot in my team because it looks easier to use, but I need this feature.
Is there a way to encode through valibot's schema?

**Comment by fabian-hiller** [comment]
Maybe the following link helps: https://valibot.dev/guides/methods/#transform

**Reply by fabian-hiller** [reply, marked as answer]
Valibot currently only allows decode and not encode. What exactly is your use case? Why do you need to encode? Can you send me Valibot pseudocode with your dream API. Then I can think about the implementation.

But in principle, due to the modularity of Valibot, you can also write your own method that adds functionality that Valibot doesn't currently have.

**Reply by rjcnd105** [reply]
I couldn't find the information I was looking for in the link you sent.
What I want is to write a transform corresponding to deparse that takes a shape corresponding to the output and spits out a shape corresponding to the input.

Please check the encoding code from the link below.

https://github.com/Effect-TS/schema#parsing-from-nullable-fields

**Reply by rjcnd105** [reply]
I'm a bit late in checking comment.

**Reply by rjcnd105** [reply]
One of the use cases is to receive data through the API, parse it in the form we want to use, use it, and send it back to the server through deparse.

**Reply by fabian-hiller** [reply]
Ok. Feel free so send me Valibot pseudocode with your dream API. Then I can think about the implementation.
